backend/src/nodes/utils/ncnn_model.py

```diff
@@ -0,0 +1,524 @@
+            num_input = weight_data_length // num_filters // kernel_w // kernel_h  # type: ignore
+            shape = (num_filters, num_input, kernel_w, kernel_h)
+
+            weight_data = np.frombuffer(binf.read(weight_data_size), dtype)
+            weight_data = weight_data.reshape(shape)  # type: ignore
+            weight_dict["weight"] = NcnnWeight(weight_data, quantize_tag)
+
+            if has_bias:
+                bias_data_size = num_filters * 4
+                bias_data = np.frombuffer(binf.read(bias_data_size), np.float32)  # type: ignore
+                weight_dict["bias"] = NcnnWeight(bias_data)
+
+        return weight_dict
+
+    def load_model(self, param_path: str, bin_path: Union[str, None] = None) -> None:
+        if bin_path is None:
+            bin_path = param_path.replace(".param", ".bin")
+
+        assert os.path.exists(param_path), f"{param_path} does not exist"
+        assert os.path.exists(bin_path), f"{bin_path} does not exist"
+
+        with open(param_path, "r") as paramf:
+            with open(bin_path, "rb") as binf:
+                paramf.readline()
+                counts = paramf.readline().strip().split(" ")
+                self.node_count = int(counts[0])
+                self.blob_count = int(counts[1])
+
+                for line in paramf:
+                    op_type, layer = self.parse_param_layer(line)
+                    layer.weight_data = self.load_layer_weights(binf, op_type, layer)
+                    self.add_layer(layer)
+
+    def write_param(self, filename: str) -> None:
+        with open(filename, "w", encoding="utf-8") as f:
+            f.write(f"{self.MAGIC}\n")
+            f.write(f"{self.node_count} {self.blob_count}\n")
+
+            for layer in self.layer_list:
+                layer_str = (
+                    f"{layer.op_type:<16} "
+                    f"{layer.name:<24} "
+                    f"{layer.num_inputs} "
+                    f"{layer.num_outputs} "
+                    f"{self.stringify_list(layer.inputs)}"
+                    f"{self.stringify_list(layer.outputs)}"
+                    f"{str(layer.params)}".rstrip()
+                )
+                f.write(layer_str + "\n")
+
+    def write_param_to_mem(self) -> str:
+        param_str = f"{self.MAGIC}\n{self.node_count} {self.blob_count}\n"
+        for layer in self.layer_list:
+            layer_str = (
+                f"{layer.op_type:<16} "
+                f"{layer.name:<24} "
+                f"{layer.num_inputs} "
+                f"{layer.num_outputs} "
+                f"{self.stringify_list(layer.inputs)}"
+                f"{self.stringify_list(layer.outputs)}"
+                f"{str(layer.params)}".rstrip()
+            )
+            param_str += layer_str + "\n"
+
+        return param_str
+
+    def write_bin(self, filename: str) -> None:
+        with open(filename, "wb") as f:
+            f.write(self.weights_bin)
+
+    def interpolate_ncnn(self, model_b, alpha):
```

Comment:
Types please.
```suggestion
    def interpolate_ncnn(self, model_b: "NcnnModel", alpha: float) -> "NcnnModel":
```

Reply:
Is that correct? Why did you surround the type with quotes?

Reply:
Because Python can't handle self-referential types. Remember, types aren't static in python, they are a runtime construct. So I can't reference the class `NcnnModel` until after it is defined. Quotes are the only way around this.

Reply:
(deleted the original, so reposting) 
Why did you put this in quotes?

Reply:
Oh interesting. I left it untyped because it caused an error. Didn't realize you could get around that with quotes.